utils/writeable/src/lib.rs

```diff
@@ -340,12 +335,22 @@ macro_rules! impl_display_with_writeable {
     };
 }
 
-/// Testing macros for types implementing Writeable. The first argument should be a
-/// `Writeable`, the second argument a string, and the third argument (*_parts_eq only)
-/// a list of parts (`[(usize, usize, Part)]`).
+/// Testing macros for types implementing [`Writeable`].
 ///
-/// The macros tests for equality of string content, parts (*_parts_eq only), and
-/// verify the size hint.
+/// Arguments, in order:
+///
+/// 1. The [`Writeable`] under test
+/// 2. The expected string value
+/// 3. [`*_parts_eq`] only: a list of parts (`[(start, end, Part)]`)
```

Comment:
```suggestion
/// 3. [`*_parts_eq`] only: a list of parts (e.g. `[(0, 7, Part::START)]`)
```

Reply:
I wanted to be more clear that `0, 7` means "start, end" because otherwise they're just numbers.

Reply:
it's not clear that `start` and `end` are meant to be numbers though

Reply:
I think the docstest illustrates that

Reply:
It's 50 lines in but whatever